prime-router/src/main/kotlin/ReportStreamFilter.kt

```diff
@@ -0,0 +1,97 @@
+    JURISDICTIONAL_FILTER("jurisdictionalFilter"),
+    QUALITY_FILTER("qualityFilter"),
+    ROUTING_FILTER("routingFilter");
+
+    // Reflection, so that we can write a single routine to handle all types of filters.
+    val filterProperty = ReportStreamFilters::class.memberProperties.first { it.name == this.field }
+        as KProperty1<ReportStreamFilters, ReportStreamFilter?>
+    val receiverFilterProperty = Receiver::class.memberProperties.first { it.name == this.field }
+        as KProperty1<Receiver, ReportStreamFilter>
+}
+
+/**
+ * The set of filtering objects available per Organization or per Receiver
+ * (and someday, per Sender too?)
+ *
+ * Examples of FilterTypes that can be applied:
+ *  @param jurisdictionalFilter - used to limit the data received or sent by geographical region
+ *  @param qualityFilter - used to limit the data received or sent, by quality
+ *  @param routingFilter - used to limit the data received or sent, by who sent it.
+ * We allow a different set of filters per [topic]
+ */
+data class ReportStreamFilters(
+    val topic: String,
+    val jurisdictionalFilter: ReportStreamFilter?,
+    val qualityFilter: ReportStreamFilter?,
+    val routingFilter: ReportStreamFilter?,
+) {
+
+    companion object {
+
+        // For each 'topic' that ReportStream handles, there is a set of default filters.
+        // Currently these are defined here in code.
+        // Each Organization and Receiver can override the defaults.
+        // todo move all of these to a GLOBAL Setting in the settings table
+        val defaultCovid19QualityFilter: ReportStreamFilter = listOf(
+            // valid basic test info
+            "hasValidDataFor(message_id)",
+            "hasValidDataFor(equipment_model_name)",
+            "hasValidDataFor(specimen_type)",
+            "hasValidDataFor(test_result)",
+            // valid basic human info
+            "hasValidDataFor(patient_last_name, patient_first_name)",
+            "hasValidDataFor(patient_dob)",
+            // has minimal valid location or other contact info (for contact tracing)
+            "hasAtLeastOneOf(patient_street,patient_zip_code,patient_phone_number,patient_email)",
+            // has valid date (for relevance/urgency)
+            "hasAtLeastOneOf(order_test_date,specimen_collection_date_time,test_result_date)",
+            // has at least one valid CLIA
+            "isValidCLIA(testing_lab_clia,reporting_facility_clia)",
+            // never send T (Training/Test) or D (Debug) data to the states.
+            "doesNotMatch(processing_mode_code,T,D)",
+        )
+        private val defaultCovid19Filters = ReportStreamFilters(
+            topic = "covid-19",
+            jurisdictionalFilter = listOf("allowNone()"), // Receiver *must* override this to get data!
+            qualityFilter = defaultCovid19QualityFilter,
+            routingFilter = listOf("doesNotMatch(processing_mode_code, T, D)"), // Default is no Training/Debug data
```

Comment:
Note the new default jurisdictionalFilter is to allow zilch.  This is a good safety/quality improvement.
Note that the routingFilter defaults to NOT routing Training and Debug data.   Nice to not have to specify this over and over.  And it didn't really belong in the quality filter.  This now allows you to safely have a secondary feed that is 'reverseTheQualityFilter=true' without mistakenly getting 'T' and 'D' data as well.  Another safety/quality improvement.